backend/main-service/src/main/java/kkakka/mainservice/coupon/application/DiscountService.java

```diff
@@ -0,0 +1,108 @@
+    final private CategoryRepository categoryRepository;
+
+    /* 할인 등록 */
+    @Transactional
+    public Long createDiscount(DiscountRequestDto discountRequestDto) {
+        if (discountRequestDto.isValidDate() && discountRequestDto.isValidDiscount()) {
+            if (DiscountType.PRODUCT_DISCOUNT.equals(discountRequestDto.getDiscountType())) {
+                Product product = getProduct(discountRequestDto);
+                Discount discount = Discount.create(
+                    product,
+                    discountRequestDto.getName(),
+                    discountRequestDto.getDiscount(),
+                    discountRequestDto.getStartedAt(),
+                    discountRequestDto.getExpiredAt());
+                product.changeDiscount(discountRequestDto.getDiscount());
+                productRepository.save(product);
+                discountRepository.save(discount);
+                return discount.getId();
+            }
+
+            if (DiscountType.CATEGORY_DISCOUNT.equals(discountRequestDto.getDiscountType())) {
+                Category category = getCategory(discountRequestDto);
+                Discount discount = Discount.create(
+                    category,
+                    discountRequestDto.getName(),
+                    discountRequestDto.getDiscount(),
+                    discountRequestDto.getStartedAt(),
+                    discountRequestDto.getExpiredAt());
+                List<Product> products = categoryRepository.findProductsByCategoryId(category.getId());
+                for (Product product : products) {
+                    product.changeDiscount(discountRequestDto.getDiscount());
+                    productRepository.save(product);
+                }
+                discountRepository.save(discount);
+                return discount.getId();
+            }
+        }
+        throw new KkaKkaException();
+    }
+
+    private Product getProduct(DiscountRequestDto discountRequestDto) {
+        return productRepository.findById(discountRequestDto.getProductId())
+            .orElseThrow(IllegalArgumentException::new);
+    }
+
+    private Category getCategory(DiscountRequestDto discountRequestDto) {
+        return categoryRepository.findById(discountRequestDto.getCategoryId())
+            .orElseThrow(IllegalArgumentException::new);
+    }
+
+    /* 할인 삭제 */
+    @Transactional
+    public void deleteDiscount(Long discountId) {
+        Discount discount = discountRepository.findById(discountId)
+            .orElseThrow(KkaKkaException::new);
+        if (DiscountType.PRODUCT_DISCOUNT.equals(discount.getDiscountType())) {
+            Product product = productRepository.findById(discount.getProductId())
+                .orElseThrow(KkaKkaException::new);
+            discount.deleteDiscount();
+            product.deleteDiscount();
+            discountRepository.save(discount);
+            productRepository.save(product);
+        }
```

Comment:
```suggestion
            productRepository.save(product);
            return;
        }
```
 지금 코드라면 아래 if 문을 수행하게 되니까 early return 해주세요!